app/transformers/PAYETransformer.scala

```diff
@@ -0,0 +1,300 @@
+}
+
+object PAYETransformer extends ServicesConfig {
+  protected def mode: Mode = Play.current.mode
+
+  protected def runModeConfiguration: Configuration = Play.current.configuration
+
+  val serviceUrl = baseUrl("tax-summaries-hod")
+
+  def middleTierJson(nino: String, year: Int): JsObject =
+    Json.obj( // add nino
+      "taxYear" -> JsNumber(year),
+      "nino"    -> JsString(nino),
+      "lnks" -> Json.arr(
+        Json.obj(
+          "rel"  -> "self",
+          "href" -> s"$serviceUrl/individuals/annual-tax-summary/$nino/$year"
+        )),
+      "allowance_data"     -> Json.obj("payload" -> Json.obj()),
+      "capital_gains_data" -> Json.obj("payload" -> Json.obj()),
+      "income_data"        -> Json.obj("payload" -> Json.obj()),
+      "income_tax"         -> Json.obj("payload" -> Json.obj(), "rates" -> Json.obj()),
+      "summary_data"       -> Json.obj("payload" -> Json.obj(), "rates" -> Json.obj()),
+      "gov_spending"       -> Json.obj()
+    )
+
+  implicit class RichJsObject(original: JsObject) extends PAYETransformer {
+    def omitEmpty: JsObject = original.value.foldLeft(original) {
+      case (obj, (key, JsString(st))) if st.isEmpty                                                   => obj - key
+      case (obj, (key, JsArray(arr))) if arr.isEmpty                                                  => obj - key
+      case (obj, (key, JsObject(child))) if new RichJsObject(JsObject(child)).omitEmpty.value.isEmpty => obj - key
+      case (obj, (_, _))                                                                              => obj
+    }
+
+    def safeTransform(jsonTransformer: Reads[JsObject]): JsObject =
+      original.transform(jsonTransformer).asOpt match {
+        case Some(json) => json
+        case None       => original
+      }
+
+    def transformPaye(source: JsObject): JsObject =
+      transformTotalIncome(source)
+        .transformSummary(source)
+        .transformAllowances(source)
+        .transformIncomeTax(source)
+        .transformGovSpendingData(source)
+        .omitEmpty
+
+    def transformTotalIncome(source: JsObject): JsObject = {
+      val income: Option[Double] = pickAmount(__ \ 'income \ 'incomeFromEmployment, source)
+      val statePension: Option[Double] = pickAmount(__ \ 'income \ 'statePension, source)
+      val otherPensionIncome: Option[Double] = pickAmount(__ \ 'income \ 'otherPensionIncome, source)
+      val otherIncome: Option[Double] = pickAmount(__ \ 'income \ 'otherIncome, source)
+      val incomeBeforeTax: Option[Double] = pickAmount(__ \ 'income \ 'incomeBeforeTax, source)
+      val employmentBenefits: Option[Double] = pickAmount(__ \ 'income \ 'employmentBenefits, source)
+      val taxableStateBenefits: Option[Double] = pickAmount(__ \ 'taxableStateBenefits, source)
+
+      val jsonTransformer =
+        appendAttribute(
+          __ \ 'income_data \ 'payload,
+          middleTierAmountJson("income_from_employment", income.getOrElse(0))) andThen
```

Comment:
You can refactor middleTierAmountJson to take in a Option[Double] and default it to zero in that function, it would remove all these getOrElse statements

the other option is to change pickAmount to return a Double and default it to zero.